bestbook/api/auth.py

```diff
@@ -14,30 +14,56 @@ def is_admin(username):
     return '/people/%s' % username in admins
 
 
-def login(email, password):
-    """Authenticates user and returns session"""
-
-    email = email.replace(' ', '+')
-
-    # We're already logged in, so return a True value
-    if session.get('username'):
-        return session.get('username')
+def authenticate(
+    email: str = "", password: str = "", s3_keys: Optional[Dict] = None, expected_username: str = ""
```

Comment:
`from typing import Dict, Optional`  # needs to be added in the imports section at the top of this file.